### PR TITLE
feat(`participation`): add `registrationPeriod` and `withdrawalDelay` params

### LINKED ISSUE #573: Investigate on the delay for withdrawing allocations
We specified using a param `WithdrawalAllocationdelay` for the delay when allocations for an auction can be withdrawn (after the auction is finished). In our current design, a user may undelegate stake and delegate it with another account for having new allocations, so it may make sense to set the allocation withdrawal delay to the unbonding period independently on if the auction is finished.

This task is about doing the investigation on what is the best way to represent the delay before allocations for an auction can be withdrawn and implementing the structure to be able to determine the delay (like for example adding the `WithdrawalAllocationdelay`)

Depends on https://github.com/tendermint/spn/issues/569

### PR DESCRIPTION
**Pull request by giunatale**
Closes #573

### What does this PR does?

Adds `registrationPeriod` and `withdrawalDelay` params based on discussion done in #573.

### CODE REVIEW

**Review by aljo242 — APPROVED** [review]

File: config.yml [hunk 1]
```diff
@@ -77,6 +77,8 @@ genesis:
             requiredAllocations: "10"
             benefits:
               maxBidAmount: "30000"
+        registrationPeriod: "604800"
+        withdrawalDelay: "1209600"
```
Comment (lubtd) [inline config.yml:81]: We should add comments to precise what it represent
Is `registrationPeriod == 1/3 unbondingPeriod` and `withdrawalDelay == 2/3 unbondingPeriod` ?

File: x/participation/types/params.go [hunk 2]
```diff
@@ -3,19 +3,25 @@ package types
...
 var _ paramtypes.ParamSet = (*Params)(nil)
 
 var (
-	KeyAllocationPrice           = []byte("AllocationPrice")
-	KeyParticipationTierList     = []byte("ParticipationTierList")
-	KeyWithdrawalAllocationDelay = []byte("WithdrawalAllocationDelay")
+	// MinTimeframeSize defines the minimum allowed value for either `RegistrationPeriod` or `WithdrawalDelay`
+	MinTimeframeSize = int64(time.Hour.Seconds() * 24) // One day
+	// MaxTimeframeSize defines the maximum allowed value for either `RegistrationPeriod` or `WithdrawalDelay`
+	MaxTimeframeSize = int64(time.Hour.Seconds() * 24 * 21) // 21 days
```
Comment (lubtd) [inline x/participation/types/params.go:19]: We hardcode here the Unbonding Period while it may change with governance or be different than 21 days.

In my humble opinion, we don't need to add such restriction in the code. We can just set the `registrationPeriod` and `withdrawalDelay` params accordingly in the idea that you can't unbonding tokens to participate in other fundraising.
Eventually when someone want to make a governance proposal to make a change in these params, they must justify the new values and if out of the bound of unbonding period, the community has the responsibility to approve or not these changes

Comment (lubtd) [reply x/participation/types/params.go:19]: IMO the only restriction we should have is that both values are positive
We can't really predict the usage of the platform, if auctions are very successful only few hours for registration may be necessary

Comment (giunatale) [reply x/participation/types/params.go:19]: This was not intended to restrict to the unbonding period but just to prevent unreasonable values. They way I have set these constants is in fact arbitrary. But yeah it makes sense to remove them entirely

File: config.yml (hunk 1 above)
Comment (giunatale) [reply config.yml:81]: well, yes, considering the unnbonding period is 21 days. I'll add comments

**Review by aljo242 — APPROVED** [review]

**Comment by lubtd** [comment]
> This was not intended to restrict to the unbonding period but just to prevent unreasonable values. They way I have set these constants is in fact arbitrary. But yeah it makes sense to remove them entirely

Also for testing to can make sense to have the ability to have very short values

File: x/participation/types/params.go [hunk 3]
```diff
@@ -50,9 +56,8 @@ var (
 			},
 		},
 	}
-
-	// TODO: Determine the default values for this param
-	DefaultWithdrawalAllocationDelay = uint64(0)
+	DefaultRegistrationPeriod = int64(time.Hour.Seconds() * 24 * 7)  // One week
+	DefaultWithdrawalDelay    = int64(time.Hour.Seconds() * 24 * 14) // Two weeks
```
[the remark this answers is not in the document]
Comment (lubtd) [reply x/participation/types/params.go:55]: @giunatale any thought on this suggestion?

[the remark this answers is not in the document]
Comment (giunatale) [reply x/participation/types/params.go:55]: something brief that makes sense, I could come up with something. But I'd rather explain in more detail in the proper docs.

File: x/participation/types/params_test.go [hunk 4]
```diff
@@ -203,3 +207,37 @@ func TestValidateTierBenefits(t *testing.T) {
 		})
 	}
 }
+
```
[the remark this answers is not in the document]
Comment (giunatale) [reply x/participation/types/params_test.go:210]: Can you rephrase? I didn't understand what you are asking here

[the remark this answers is not in the document]
Comment (lubtd) [reply x/participation/types/params_test.go:210]: `(Params) Validate()` validate the whole parameter structure. We should assert that this method fails if `validateAllocationsUsageTimeFrame` fails for `registrationPeriod ` or `withdrawalDelay `

[the remark this answers is not in the document]
Comment (lubtd) [reply x/participation/types/params_test.go:210]: Basically I'm suggesting having two more test case in `TestParamsValidate` with negative `registrationPeriod` and `withdrawalDelay`

[the remark this answers is not in the document]
Comment (giunatale) [reply x/participation/types/params_test.go:210]: That actually made me notice `(Params) Validate()` didn't have validation for these two. I'll add it. It's no issue for me adding the two more test cases also, I was just wondering if it goes against the code netiquette we are adopting.

**Review by lubtd — COMMENTED** [review]
We have some test failure apparently

**Comment by aljo242** [comment]
The unit test passes now, the fail is because the upload to CodeCov had some error

**Review by lubtd — APPROVED** [review]
